contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/HdfsCopyStats.java

```diff
@@ -0,0 +1,226 @@
+        } catch(Exception e) {
+            statsFileWriter = null;
+            logger.error("Error during stats directory for destination " + destination, e);
+            return;
+        }
+    }
+
+    public HdfsCopyStats(String source,
+                         File destination,
+                         boolean enableStatsFile,
+                         int maxVersionsStatsFile,
+                         boolean isFileCopy,
+                         long totalBytes) {
+        this.sourceFile = source;
+        this.totalBytesCopied = 0L;
+        this.bytesSinceLastReport = 0L;
+        this.totalBytes = totalBytes;
+        this.lastReportNs = System.nanoTime();
+        this.startTimeMS = System.currentTimeMillis();
+        initializeStatsFile(destination, enableStatsFile, maxVersionsStatsFile, isFileCopy);
+    }
+
+    public void recordBytes(long bytes) {
+        this.totalBytesCopied += bytes;
+        this.bytesSinceLastReport += bytes;
+    }
+
+    public void reset() {
+        this.bytesSinceLastReport = 0;
+        this.lastReportNs = System.nanoTime();
+    }
+
+    public long getBytesSinceLastReport() {
+        return bytesSinceLastReport;
+    }
+    
+    private void reportStats(String message) {
+        try {
+            if(statsFileWriter != null) {
+                statsFileWriter.write(dateFormat.format(new Date()));
+                statsFileWriter.write(",");
+                statsFileWriter.write(message);
+                statsFileWriter.newLine();
+            }
+        } catch(IOException e) {
+            // ignore errors
+        }
+
+    }
+
+    public void reportError(String message, Throwable t) {
+        if(statsFileWriter != null && t != null) {
+            reportStats(message + " Error Message : " + t.getMessage());
+            PrintWriter pw = new PrintWriter(statsFileWriter);
+            t.printStackTrace(pw);
+        }
+    }
+
+    public void reportFileError(File file,int attempts, long startTimeMS, Throwable t) {
+        long nowMS = System.currentTimeMillis();
+        String message = " Error occured during file download " + file.getName()
+                         + " after attempts " + attempts + " time elapsed " + (nowMS - startTimeMS)
+                         + " MS.";
+        reportError(message, t);
+    }
+
+    public void reportFileDownloaded(File file,
+                                     long startTimeMS,
+                                     long fileSize,
+                                     long timeTakenMS,
+                                     int attempts,
+                                     long totalBytesDownloaded) {
+        reportStats(file.getName() + "," + startTimeMS + "," + fileSize + "," + timeTakenMS + ","
+                    + attempts + "," + totalBytesDownloaded);
+    }
+
+    public void complete() {
+        long nowMS = System.currentTimeMillis() ;
+        reportStats(" Completed at " + nowMS + "MS. Total bytes Copied " + totalBytesCopied
```

Comment:
Human-readable time would be nice. If you want to also include epoch time for easier parsing by scripts, I'm fine with it... as you wish.